### PR TITLE
Ent - Dependency management

### PR DESCRIPTION
**Pull request by mrizzi**
# Description of the PR

Provide implementation for endpoints related to `Dependency` management.

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [x] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/assembler/backends/ent/dependency.go [hunk 1]
```diff
@@ -42,13 +44,15 @@ type Dependency struct {
...
-	// DependentPackage holds the value of the dependent_package edge.
-	DependentPackage *PackageName `json:"dependent_package,omitempty"`
+	// DependentPackageName holds the value of the dependent_package_name edge.
+	DependentPackageName *PackageName `json:"dependent_package_name,omitempty"`
+	// DependentPackageVersion holds the value of the dependent_package_version edge.
+	DependentPackageVersion *PackageVersion `json:"dependent_package_version,omitempty"`
 	// loadedTypes holds the information for reporting if a
 	// type was loaded (or requested) in eager-loading or not.
-	loadedTypes [2]bool
+	loadedTypes [3]bool
```
Comment (mihaimaruseac) [inline pkg/assembler/backends/ent/dependency.go:53]: Question for my understanding: What is the reason for the change from 2 to 3 here?

Comment (mihaimaruseac) [reply pkg/assembler/backends/ent/dependency.go:53]: I see line 87 below. Could we instead make these bools separate variables with clear names?

File: pkg/assembler/backends/ent/dependency.go [hunk 2]
```diff
@@ -64,25 +68,38 @@ func (e DependencyEdges) PackageOrErr() (*PackageVersion, error) {
...
+		return e.DependentPackageName, nil
 	}
-	return nil, &NotLoadedError{edge: "dependent_package"}
+	return nil, &NotLoadedError{edge: "dependent_package_name"}
+}
+
+// DependentPackageVersionOrErr returns the DependentPackageVersion value or an error if the edge
+// was not loaded in eager-loading, or loaded but was not found.
+func (e DependencyEdges) DependentPackageVersionOrErr() (*PackageVersion, error) {
+	if e.loadedTypes[2] {
```
Comment (mihaimaruseac) [inline pkg/assembler/backends/ent/dependency.go:87]: This seems very likely to cause issues on future refactoring as it is unclear why the element at index 2 is the one to be used here

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: pkg/assembler/backends/ent/dependency.go (hunk 1 above)
Comment (mrizzi) [reply pkg/assembler/backends/ent/dependency.go:53]: This code is generated by Ent so I'm not so sure we have room for customization.
At the same time, the fact that they're generated from Ent could also mean we won't have to work on them directly so having unclear names isn't a big problem.

WDYT?

File: pkg/assembler/backends/ent/dependency.go (hunk 2 above)
Comment (mrizzi) [reply pkg/assembler/backends/ent/dependency.go:87]: The same consideration as above applies here.

File: pkg/assembler/backends/ent/dependency.go (hunk 1 above)
Comment (mihaimaruseac) [reply pkg/assembler/backends/ent/dependency.go:53]: Makes sense, I didn't realzie that. Resolving

**Review by mihaimaruseac — APPROVED** [review]

**Review by jeffmendoza — APPROVED** [review]
